src/m1_facilitator/lib/TransactionExecutor.ts

```diff
@@ -12,17 +12,155 @@
+    this.gasPrice = gasPrice;
+    this.avatarAccount = avatarAccount;
+    this.pollInterval = 1 * 60 * 1000; // in ms
+    this.setIntervalHandle = null;
+  }
+
+  /**
+   * This method enqueues transactions to queue. TransactionRepository acts as
+   * queue here. The method is called by services.
+   *
+   * @param toAddress Contract address at which transaction needs to be sent.
+   * @param rawTx Raw transaction object.
+   */
+  public async add(toAddress: string, rawTx: TransactionObject<string>): Promise<void> {
+    const transaction = new Transaction(
+      this.avatarAccount.address,
+      toAddress,
+      rawTx.encodeABI(),
+      this.gasPrice,
+    );
+    await this.transactionRepository.save(transaction);
+  }
+
+  /**
+   * It starts processing pending transactions in regular intervals.
+   */
+  public async start(): Promise<void> {
+    await this.execute();
+    this.setIntervalHandle = setInterval(
+      async (): Promise<void> => this.execute(),
+      this.pollInterval,
+    );
+  }
+
+  /**
+   * Stops processing of transactions by clearing setIntervalHandle.
+   * It should be called when facilitator stops.
+   */
+  public async stop(): Promise<void> {
+    if (this.setIntervalHandle) {
+      clearInterval(this.setIntervalHandle);
+    }
+  }
+
+  /**
+   * Execute method does below:
+   * - Dequeues transaction record in regular interval
+   * - Process the transaction and constructs data for transaction to be executed
+   * - Sends the transaction by calling sendTransaction method.
+   * - Updates Transaction repository with information like transactionHash, gas, nonce
+   */
+  private async execute(): Promise<void> {
+    const transaction = await this.transactionRepository.dequeue();
+    const nonce = await this.avatarAccount.getNonce(this.web3);
+    if (transaction !== null) {
+      const txHash = await this.sendTransaction(transaction, nonce);
+      transaction.transactionHash = txHash;
+      transaction.gas = this.gas;
+      transaction.nonce = nonce;
+    }
+  }
+
   /**
-   * Method adds a raw transaction to queue.
-   * @param rawtx Raw transaction object.
+   * This method submits a raw transaction and returns transaction hash.
+   *
+   * @param transaction Transaction repository record.
+   * @param nonce Avatar account nonce.
+   *
+   * @return Transaction hash.
    */
-  // @ts-ignore
-  public async add(rawtx: TransactionObject<string>): Promise<void> {
+  private async sendTransaction(transaction: Transaction, nonce: BigNumber): Promise<string> {
+    Logger.info(`Transaction to be processed: ${JSON.stringify(transaction)}, nonce: ${nonce.toString(10)}`);
+    return new Promise(async (onResolve, onReject): Promise<void> => {
+      let txOptions = {
+        from: transaction.fromAddress,
+        to: transaction.toAddress,
+        data: transaction.encodedData,
+        nonce: nonce.toNumber(),
+        gas: transaction.gas ? transaction.gas.toString() : transaction.gas,
```

Comment:
This line is confusing, any better way to do this?

Reply:
Updated with below:
```
transaction.gas && transaction.gas.toString()
```
if transaction.gas value exists and then only do transaction.gas.toString()